react/features/device-selection/components/DeviceSelectionDialog.js

```diff
@@ -421,32 +308,26 @@ class DeviceSelectionDialog extends Component {
 
         this._isClosing = true;
 
-        const deviceChangePromises = [];
-
-        if (this.state.videoInput && !this._isPreviewingCurrentVideoTrack()) {
-            const changeVideoPromise = this._disposeVideoPreview()
-                .then(() => {
-                    this.props.dispatch(setVideoInputDevice(
-                        this.state.videoInput));
-                });
-
-            deviceChangePromises.push(changeVideoPromise);
-        }
-
-        if (this.state.audioInput && !this._isPreviewingCurrentAudioTrack()) {
-            const changeAudioPromise = this._disposeAudioPreview()
-                .then(() => {
-                    this.props.dispatch(setAudioInputDevice(
-                        this.state.audioInput));
-                });
-
-            deviceChangePromises.push(changeAudioPromise);
-        }
-
-        if (this.state.audioOutput
-            && this.state.audioOutput !== this.props.currentAudioOutputId) {
-            this.props.dispatch(setAudioOutputDevice(this.state.audioOutput));
-        }
+        const deviceChangePromises = this._attemptPreviewTrackCleanup()
+            .then(() => {
+                if (this.state.selectedVideoInputId
```

Comment:
I'm fairly certain this was another cross browser concern. I left in my personal notes that one browser was throwing an error when disposing the preview track while existing device switch logic was creating a new track for use. In the future, it's probably better for the preview track to be just re-used anyway, but I think I wanted to avoid touching existing device switch logic at the time.